src/io/flutter/logging/FlutterLogEntryParser.java

```diff
@@ -101,15 +102,29 @@ private FlutterLogEntry parseLoggingEvent(@NotNull Event event) {
       }
     }
 
-    // TODO(pq): If message.getValueAsStringIsTruncated() is true, we'll need to retrieve the full string
-    // value and update this entry after creation.
-    String messageStr = message.getValueAsString();
+    final String messageStr = message.getValueAsString();
+    final FlutterLogEntry entry = lineHandler.parseEntry(messageStr, category, level);
+
     if (message.getValueAsStringIsTruncated()) {
-      messageStr += "...";
-    }
+      entry.setMessage(messageStr + "...");
+      final Isolate isolate = new Isolate(json.get("isolate").getAsJsonObject());
+      debugProcess.getVmServiceWrapper().getObject(isolate.getId(), message.getId(), new GetObjectConsumer() {
+        @Override
+        public void received(Obj response) {
+          entry.setMessage(((Instance)response).getValueAsString());
```

Comment:
FYI this can happen in a different thread. Make sure you trampoline back to the UI thread.

Reply:
Additionally, we also need to schedule an event to repaint the UI. This code currently appears to be just assuming another UI repaint will happen.